program-test/src/lib.rs

```diff
@@ -698,3 +729,59 @@ impl ProgramTestBanksClientExt for BanksClient {
         ))
     }
 }
+
+struct DroppableTask<T>(Sender<()>, JoinHandle<T>);
+
+impl<T> Drop for DroppableTask<T> {
+    fn drop(&mut self) {
+        if let Err(e) = self.0.try_send(()) {
+            debug!("Error cancelling task: {:?}", e);
+        }
+    }
+}
+
+pub struct ProgramTestContext {
+    pub banks_client: BanksClient,
+    pub payer: Keypair,
+    pub last_blockhash: Hash,
+    _bank_task: DroppableTask<()>,
+}
+
+impl ProgramTestContext {
+    fn new(
+        bank_forks: Arc<RwLock<BankForks>>,
+        banks_client: BanksClient,
+        payer: Keypair,
+        last_blockhash: Hash,
+        genesis_config: GenesisConfig,
+    ) -> Self {
+        // Run a simulated PohService to provide the client with new blockhashes.  New blockhashes
+        // are required when sending multiple otherwise identical transactions in series from a
+        // test
```

Comment:
This behaviour is important for the user to know about and it won't occur if the user uses `start()`.  How about a the function header comment for `start_with_context()` that mentions this and how `start()` doesn't have it.   This can then for the basis for additional new features that are only supported by `start_with_context()`.  Then in v1.6 timeframe we could even consider marking `start()` as deprecated

Reply:
I'm not sure I understand -- lines 658-670 start up the ticking task for `start()`, so do you want to get rid of that to differentiate with `start_with_context()`?

Reply:
Ah, sorry got confused there for a moment.  My comment makes no sense, withdrawn!